src/ResourceManager/Compute/Commands.Compute.Test/ScenarioTests/StrategiesVirtualMachineTests.cs

```diff
@@ -31,11 +37,65 @@ public void TestSimpleNewVm()
             ComputeTestController.NewInstance.RunPsTest("Test-SimpleNewVm");
         }
 
-        [Fact] 
-        [Trait(Category.AcceptanceType, Category.CheckIn)] 
+        [Fact]
+        [Trait(Category.AcceptanceType, Category.CheckIn)]
+        public void TestSimpleNewVmWithAvailabilitySet()
+        {
+            ComputeTestController.NewInstance.RunPsTest("Test-SimpleNewVmWithAvailabilitySet");
+        }
+
+        [Fact]
+        [Trait(Category.AcceptanceType, Category.CheckIn)]
         public void TestSimpleNewVmWithAvailabilitySet2()
-        { 
-            ComputeTestController.NewInstance.RunPsTest("Test-SimpleNewVmWithAvailabilitySet2"); 
+        {
+            ComputeTestController.NewInstance.RunPsTest("Test-SimpleNewVmWithAvailabilitySet2");
+        }
+
+        public static void TestDomainName(string psTest, Func<string> getUniqueId)
+        {
+            var create = typeof(UniqueId).GetField("_Create", BindingFlags.Static | BindingFlags.NonPublic);
+            var oldCreate = create.GetValue(null);
+            try
+            {
+                // mock UniqueId.Create()
+                create.SetValue(null, getUniqueId);
+
+                var callingClassType = TestUtilities.GetCallingClass(2);
+                var mockName = TestUtilities.GetCurrentMethodName(2);
+
+                ComputeTestController.NewInstance.RunPsTestWorkflow(
+                    () => new[] { psTest },
+                    // no custom initializer
+                    null,
+                    // no custom cleanup 
+                    null,
+                    callingClassType,
+                    mockName);
+            }
+            finally
+            {
+                create.SetValue(null, oldCreate);
+            }
+        }
+
+        [Fact]
+        [Trait(Category.AcceptanceType, Category.CheckIn)]
+        public void TestSimpleNewVmWithDefaultDomainName()
+        {
+            TestDomainName("Test-SimpleNewVmWithDefaultDomainName", () => "a469f1");
```

Comment:
I worry about runnign these tests live. f we ran these tests from two contexts, the second one would fail. I feel like we should be generating these values randomly, using the test GetAsset functionality (https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SdkCommon/TestFramework/Microsoft.Azure.Test.HttpRecorder/HttpMockServer.cs#L174 ), so that the values are stored in the test recording for playback.  I.  We could useGetAssetName or  GetAssetGuid.  We could then pass the expected value to the test case as a parameter, and thus avoid hard-coded values that can cause naming collisions.